### DISCUSSION TITLE
Should I start my new b2b project with V2?

### DISCUSSION THREAD

**Discussion by tirocoder** [medusajs/medusa#8746] [Q&A]
Hello. I want to use medusajs for a b2b project, but during the transition to v2, I couldn't decide whether I should do this project with v1 or v2. The project needs many integrations and special developments, such as adding new languages, local shipping and payment integrations and many more. Wouldn't it be very difficult, perhaps impossible, to adapt these improvements to v2?

**Comment by Deroswent** [comment]
Depends on your deadlines.  If you don't plan to launch the project into production in the next 6-10 months, you can start using V2.

But if you need to start production faster - do not use v2.  

V2 is a great product, but it is at an early stage of development and is not ready for production use yet, it contains a lot of bugs and its logic is constantly being redesigned by developers.

**Reply by docloulou** [reply]
In the first preview, the developers mentioned 3-4 months, not 6-10 months. I hope you’re wrong! I’m also waiting for a release candidate to start my e-commerce project. Fingers crossed it will be ready by the end of the year.

**Reply by olivermrbl** [reply]
@Deroswent, thanks for pitching in, but this is not correct. The release candidate is expected to be published in approx. a month. The official release is expected to be published at end of October. 

cc @izgaradusmani, @docloulou

**Reply by docloulou** [reply]
> @Deroswent, thanks for pitching in, but this is not correct. The release candidate is expected to be published in approx. a month. The official release is expected to be published at end of October.
> 
> cc @izgaradusmani, @docloulou

Thanks @olivermrbl , Medusa Team did a great job. Version 2 will be modern and easily deployable in an ERP/CRM environment. The only issue is that the documentation is lacking, but I think that's because the code is evolving rapidly. I can't wait to start tinkering with it!

**Reply by olivermrbl** [reply]
@docloulou, thanks for the kind words.

What can we do to improve the documentation? We would be very interested in feedback.

cc @shahednasser

**Reply by docloulou** [reply]
@shahednasser @olivermrbl 
 https://docs.medusajs.com/v2/basics/commerce-modules#resolve-commerce-module-services

For example, in this page, I understand how to retrieve product data in another service/route... but I'm missing the link to other references/a list of available modules (other than through VS Code, which references them with the help of TypeScript...). For instance, if I want to list orders, having a general reference/resource list would be helpful.

Another note: I might be a bit demanding, but while waiting for other official modules, we don't have many code examples, or at least not enough precise ones.

**Reply by shahednasser** [reply]
Hi @docloulou thank you for the feedback!

That documentation page points to a link of all commerce modules at an earlier point (which is [here](https://docs.medusajs.com/v2/resources/commerce-modules)). However, if you missed it then we definitely need to look into why you missed it and improve on that.

If you refer to that documentation, you'll find a reference for each module's service with its methods and how to use them.

> Another note: I might be a bit demanding, but while waiting for other official modules, we don't have many code examples, or at least not enough precise ones.

I'm not sure I understand this part. Can you elaborate? would love to hear more feedback that you may have.

**Reply by tirocoder** [reply]
> @Deroswent, thanks for pitching in, but this is not correct. The release candidate is expected to be published in approx. a month. The official release is expected to be published at end of October.
> 
> cc @izgaradusmani, @docloulou

Thanks @olivermrbl. October is a good date. We will be waiting with excitement. We can carry out the project to go live after October. But still, does it make sense to start developing in the v2 environment and improve the project by getting updates, or would it be wise to start with v1 and then do a migration project?
